distributed_shampoo/utils/shampoo_preconditioner_list.py

```diff
@@ -889,8 +891,65 @@ def _precondition_grad(
         )
 
 
-class ShampooPreconditionerList(
+class BaseShampooPreconditionerListBasedOnShampooPreconditionerConfig(
     BaseShampooPreconditionerList[
+        ShampooKroneckerFactorsStateType, ShampooKroneckerFactorsListType
```

Comment:
One problem with an intermediate class is that these types are incorrect -- ideally we should not specify any types here, but not sure if you can inherit from a `Generic` class without specifying the types.